core/services/workflows/delegate_test.go

```diff
@@ -6,57 +6,150 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/smartcontractkit/chainlink/v2/core/services/workflows"
+	"github.com/smartcontractkit/chainlink/v2/core/testdata/testspecs"
 )
 
 func TestDelegate_JobSpecValidator(t *testing.T) {
 	t.Parallel()
 
 	var tt = []struct {
-		name  string
-		toml  string
-		valid bool
+		name           string
+		workflowTomlFn func() string
+		valid          bool
 	}{
 		{
-			"valid spec",
-			`
-type = "workflow"
-schemaVersion = 1
-workflowId = "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
-workflowOwner = "00000000000000000000000000000000000000aa"
-workflowName = "test"
-`,
+			"missing workflow field",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				return testspecs.GenerateWorkflowSpec(id, owner, "test", "").Toml()
+			},
+			false,
+		},
+
+		{
+			"null workflow",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				return testspecs.GenerateWorkflowSpec(id, owner, "test", "{}").Toml()
+			},
+			false,
+		},
+
+		{
+			"missing name",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				wf := `
+triggers: []
+consensus: []
+targets: []
+`
+				return testspecs.GenerateWorkflowSpec(id, owner, "", wf).Toml()
+			},
+			false,
+		},
+
+		{
+			"minimal passing workflow",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				wf := `
+triggers: []
+consensus: []
+targets: []
+`
+				return testspecs.GenerateWorkflowSpec(id, owner, "test", wf).Toml()
+			},
 			true,
 		},
+
+		// Taken from jobs controller test, as we want to fail early without a db / slow test dependency
+		{
+			"valid full spec",
+			func() string {
```

Comment:
Should the job spec for the workflow spec also include the signature field? E.g.,

```
id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
signature := "00000000000000000000000000000000000000aa"
workflow := `...`
```
which is validated to be the one from the workflow spec?

Reply:
This would require supporting empty `owner_field` and `signature` fields

Reply:
I'm not too sure, I kept the structure of the test cases the same as trunk. I think we should address this in a separate topic as I want to get validation for workflow specs (not job specs) merged in first.